features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/rolesandpermissions/changeroles/ChangeRolesView.kt

```diff
@@ -189,7 +188,7 @@ fun ChangeRolesView(
                 )
             }
             is AsyncAction.Success -> {
-                SideEffect { updatedOnBackPressed() }
+                LaunchedEffect(action) { updatedOnBackPressed() }
```

Comment:
Another possible fix is to use AsyncActionView:

```kotlin
        AsyncActionView(
            async = state.exitState,
            onSuccess = { updatedOnBackPressed() },
            confirmationDialog = {
                ConfirmationDialog(
                    title = stringResource(CommonStrings.dialog_unsaved_changes_title),
                    content = stringResource(CommonStrings.dialog_unsaved_changes_description_android),
                    onSubmitClicked = { state.eventSink(ChangeRolesEvent.Exit) },
                    onDismiss = { state.eventSink(ChangeRolesEvent.CancelExit) }
                )
            },
            onErrorDismiss = { /* Cannot happen */ },
        )
```

Reply:
Oh true, it's the perfect place to use it. Thanks!